src/LumenServiceProvider.php

```diff
@@ -27,7 +27,7 @@ public function register()
     public function boot()
     {
         if ($this->app['config']->get('snappy.pdf.enabled')) {
-            $this->app['snappy.pdf'] = $this->app->share(function ($app) {
+            $this->app['snappy.pdf'] = $this->app->singleton(function ($app) {
```

Comment:
will this impact backward-compatibility of the module? maybe it can be mitigated by created a new release tag for 5.4?